### ISSUE TITLE
Odd UX: Installing an extension makes it disappear from the list

### ISSUE THREAD

**Issue by twomice** [twomice/org.civicrm.extensionsui#25]
This is not a bug, but is a little disorienting for the user.

1. Click the "Add New" tab and find an extension there that you want to install.
2. Click the "Install" link for that extension.
3. Observe the "Installing..." and "Installed" messages; then observe that the extension is immediately _desaparecido_ from the list. Of course it is, because it's installed and it no longer should be listed under "Add New". But for the user, this is a bit of a surprise. "Where's my extension? I just @#$%^ installed it, and now it's gone!"

So what's better?  Perhaps one of these options:

- After installing, switch the view to the "Installed" tab and highlight the extension in the list.
- Grey-out the newly installed extension, and keep it in the "Add New" list temporarily, as long as the "Add New" tab is continuously in view (ie., until the page is reloaded or the view is switched to the "Installed" tab.)  Implementing this option probably means changing our current approach of just reloading the "Add New" tab content; instead we would leave the tab contents as-is and just change the model to reflect a the "just-installed" status of this extension. _I like this option better._

Blocked by: #27

**Comment by universalhandle** [comment]
Agree that the current approach of reloading the content is not ideal, but seeing as how you volunteered the time to get this extension as far as it is, who am I to quibble!?

I'd like for us to refactor some of this in the not-distant future. The extension doesn't yet fully embrace the Angular paradigm, and a lot of the features we've been discussing will become low-hanging fruit when it does.

**Comment by twomice** [comment]
Yes, I'd like to see that refactoring happen sooner rather than later. You've got more Angular experience than I, so I think this will lean on you a lot. Pair programming is an option if we both like the idea and can make the time.

**Comment by twomice** [comment]
#27 describes our intent to refactor. I suggest we do that first, and then handle this one.
